extra_tests/docs_gen.py

```diff
@@ -0,0 +1,101 @@
+import sys
+import warnings
+from pydoc import ModuleScanner
+
+
+def scan_modules():
```

Comment:
Since this only concerns the builtins, why not use `sys.builtin_module_names` here?

Reply:
I found `sys.bulitin_module_names` is not covering all the c modules. I didn't investigate well but I think it is more like pre-loaded module before import.

Reply:
huh, maybe that in combination with `sys.stdlib_module_names` then (but that landed in `3.10` I believe.